hudi-client/hudi-client-common/src/test/java/org/apache/hudi/client/transaction/TestSimpleConcurrentFileWritesConflictResolutionStrategy.java

```diff
@@ -193,6 +193,38 @@ public void testConcurrentWritesWithInterleavingScheduledCompaction() throws Exc
     }
   }
 
+  @Test
+  public void testConcurrentWritesWithInterleavingInflightCompaction() throws Exception {
+    createCommit(metaClient.createNewInstantTime(), metaClient);
+    HoodieActiveTimeline timeline = metaClient.getActiveTimeline();
+    // consider commits before this are all successful
+    Option<HoodieInstant> lastSuccessfulInstant = timeline.getCommitsTimeline().filterCompletedInstants().lastInstant();
+    // writer 1 starts
+    String currentWriterInstant = metaClient.createNewInstantTime();
+    createInflightCommit(currentWriterInstant, metaClient);
+    // compaction 1 gets scheduled and becomes inflight
+    String newInstantTime = metaClient.createNewInstantTime();
+    createPendingCompaction(newInstantTime, metaClient);
+
+    Option<HoodieInstant> currentInstant = Option.of(new HoodieInstant(State.INFLIGHT, HoodieTimeline.COMMIT_ACTION, currentWriterInstant));
```

Comment:
So this is a COW table?

Reply:
Yeah, I know we should use MOR table for compaction. But this entire test uses COW to test concurrency with compaction. I guess it should be OK since we only test the conflict resolution logic.